src/main/java/org/apache/commons/codec/binary/BaseNCodecInputStream.java

```diff
@@ -39,12 +39,15 @@
 
     private final byte[] singleByte = new byte[1];
 
+    private final byte[] buf;
```

Comment:
Add a comment about a reusable buffer to avoid repeat allocations in the read method's while loop.